contracts/Core.sol

```diff
@@ -120,11 +120,15 @@ contract Core is DBC, Owned, Shares, SafeMath, CoreProtocol {
     function calcValuePerShare(uint value)
         constant
         pre_cond(notZero(totalSupply))
-        returns (uint sharePrice)
+        returns (uint valuePerShare)
     {
-        sharePrice = 10 ** decimals * value / totalSupply;
+        valuePerShare = 10 ** decimals * value / totalSupply;
     }
 
+    /// Pre: None
+    /// Post: Share price denominated in referenceAsset in baseunit of [10 ** decimals] per Share
+    function getSharePrice() constant returns (uint price){price = atLastPayout.sharePrice;}
```

Comment:
ah right it throws for empty zero shares..
atLastPayout.sharePrice can however be quite delayed
think it works for now but we should keep it in mind for later